### PR TITLE
ipatests: add a NFS test

### PR DESCRIPTION
**Pull request by fcami**
With the latest NFS changes:
- systemd NFS-related unit files
-  configuration from /etc/sysconfig/nfs to /etc/nfs.conf

testing the client configuration code is becoming paramount.

There is no need to run the NFS tests on every PR, a nightly
run on the latest Fedora release and on rawhide should be enough.
    
Fixes: https://pagure.io/freeipa/issue/7805
Signed-off-by: François Cami <fcami@redhat.com>

### CODE REVIEW

File: ipatests/test_integration/test_nfs.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+            "ipa", "host-mod", automntclt.hostname,
+            "--location", "''"
+        ])
+        # not strictly necessary, but this exercises automountlocation-del
+        self.master.run_command([
+            "ipa", "automountlocation-del", "seattle"
+        ])
+        nfsclt.run_command(["systemctl", "restart", "nfs-utils"])
+        nfssrv.run_command(["systemctl", "restart", "nfs-utils"])
+        # TODO put /etc/resolv.conf back
```
Comment (celestian) [inline ipatests/test_integration/test_nfs.py]: Is it possible to implement also this TODO?

File: ipatests/test_integration/test_nfs.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        nfssrv.run_command([
+            "ipa-getkeytab", "-p", "nfs/%s" % nfssrv.hostname,
+            "-k", "/etc/krb5.keytab"
+        ])
+        nfssrv.run_command(["systemctl", "restart", "nfs-server"])
+        nfssrv.run_command(["systemctl", "enable", "nfs-server"])
+        time.sleep(WAIT_AFTER_INSTALL)
+
+        # NFS share
+        # TODO use a better way to associate server-side exports
```
Comment (celestian) [inline ipatests/test_integration/test_nfs.py]: I am afraid this TODO will be there forever. Consider implementation or delete the comment, please.

File: ipatests/test_integration/test_nfs.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        nfsclt.run_command([
+            "systemctl", "status", "gssproxy"
+        ])
+        nfsclt.run_command([
+            "systemctl", "status", "rpc-gssd"
+        ])
+        # nfsclt.run_command([
+        #    "mount", "-t", "nfs4", "-o", "ro",
+        #    "%s:/exports/stdnfs" % nfssrv.hostname, "/mnt/std", "-v"
+        # ])
```
Comment (celestian) [inline ipatests/test_integration/test_nfs.py]: This is commented code. If it is very useful for debugging purposes, please, write one comment above it, some thing like: "For debugging." If not, just remove it, please.

Comment (fcami) [reply ipatests/test_integration/test_nfs.py]: Removed.

File: ipatests/test_integration/test_nfs.py (hunk 2 above)
Comment (fcami) [reply ipatests/test_integration/test_nfs.py]: Agreed, removed.

File: ipatests/test_integration/test_nfs.py (hunk 1 above)
Comment (fcami) [reply ipatests/test_integration/test_nfs.py]: Done. If you want to confirm, please search for resolv.conf in http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/e009d794-5fc8-11e9-b1f8-fa163e937cf0/report.html

**Comment by fcami** [comment]
Hi @celestian all comments above have been addressed.
Successful run available at http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/e009d794-5fc8-11e9-b1f8-fa163e937cf0/report.html
I have squashed the commits and am letting CI run again, will post a link to the latest report and then remove the temp commit.

**Comment by fcami** [comment]
Pushed again with larger timeouts (3600=>9000) as execution time seems to be variable.

**Comment by fcami** [comment]
Thanks @celestian for the ACK!
Post-squash tests are green: http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/fdb99e54-603c-11e9-a49b-fa163e6c17d4/
Removing temp commit.

**Comment by fcami** [comment]
Temp commit removed, waiting for CI.

**Comment by fcami** [comment]
master:

* 0ba25c7e68ae1a3a816070688eae460c6160acee ipatests: add nfs tests

**Pull request closed without merging**